mm2src/lp_ordermatch/simple_market_maker.rs

```diff
@@ -657,16 +733,32 @@ pub async fn start_simple_market_maker_bot(ctx: MmArc, req: StartSimpleMakerBotR
     let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(&ctx).unwrap();
     let mut state = simple_market_maker_bot_ctx.trading_bot_states.lock().await;
     let mut price_url = simple_market_maker_bot_ctx.price_url.lock().await;
+    let mut refresh_rate = simple_market_maker_bot_ctx.bot_refresh_rate.lock().await;
+    let mut message_service = simple_market_maker_bot_ctx.message_service.lock().await;
     match *state {
         TradingBotState::Running => MmError::err(StartSimpleMakerBotError::AlreadyStarted),
         TradingBotState::Stopping => MmError::err(StartSimpleMakerBotError::CannotStartFromStopping),
         TradingBotState::Stopped => {
+            let dispatcher_ctx = DispatcherContext::from_ctx(&ctx).unwrap();
+            let mut dispatcher = dispatcher_ctx.dispatcher.lock().await;
+            dispatcher.add_listener("lp_bot_listener", simple_market_maker_bot_ctx.clone());
             *state = TradingBotState::Running;
             drop(state);
             let mut trading_bot_cfg = simple_market_maker_bot_ctx.trading_bot_cfg.lock().await;
             *trading_bot_cfg = req.cfg;
             *price_url = req.price_url.unwrap_or_else(|| KMD_PRICE_ENDPOINT.to_string());
-            info!("simple_market_maker_bot successfully started");
+            *refresh_rate = req.bot_refresh_rate.unwrap_or(BOT_DEFAULT_REFRESH_RATE);
+            if *refresh_rate < BOT_DEFAULT_REFRESH_RATE {
+                *refresh_rate = BOT_DEFAULT_REFRESH_RATE;
+            }
+            let msg = "simple_market_maker_bot successfully started";
+            if let Some(telegram_api_key) = req.telegram_api_key {
```

Comment:
```
Also, let msg = "simple_market_maker_bot successfully started"; is not extensible. I guess we won't send such notifications in the plain text e.g. to WS or gRPC, which we will hopefully have shortly.
```

Let's talk about this in the second iteration; it looks like it's something for the Dispatcher; the pull request introduces two new objects:

A dispatcher allows delivering events and where the listener choose how to treat these events. (We can therefore imagine a Websocket context that would have a listener for specific events and decide how to handle these events).

On the other hand, we have a messaging service that focuses on delivering messages to existing/centralized services (Telegram, Whatsapp, Messenger etc ...).

If we look in `lp_bot`, I did that I subscribe to an event type (UpdateMakerEvent) and choose how to deliver the message (message_service).

I can give you an example of how to dispatch a MakerBotEvent event and use the service message to send this message. But I think it's overkill because, for the moment, we don't need to communicate in WebSocket in the bot. If it is the case, I can already go on this approach and show you what I have in mind.

Small example:

```rust
enum MakerBotEvent {
   Started
}

pub async fn start_simple_market_maker_bot(ctx: MmArc, req: StartSimpleMakerBotRequest) -> StartSimpleMakerBotResult {
    ...
      dispatcher
                        .dispatch_async(MakerBotEvent::Started)
                        .await;
} 
```

listener:

```
async fn process_event_async(&self, event: Self::Event) {
        match event {
             LpEvents::MakerBotEvent(bot_event) => {
                   // use self.ctx to retrieve message service and deliver a message to telegram
                   // use self.ctx to retrieve websocket context and send data through network.
             }
        }
}
```

Reply:
> Let's talk about this in the second iteration; it looks like it's something for the Dispatcher;

It's not for the dispatcher - Rust encourages us to use enums when possible in such cases in general. I prefer to see MakerBotEvent in this PR as it's quite common that such changes requests can be easily forgotten afterward. And I also prefer to not see plain text for something like this on future reviews. Please dedicate your time for appropriate refactoring before opening a PR and asking for a review.

Reply:
> It's a good point, which makes this topic even more controversial to me 🙂 Having separate notification settings (e.g. chat ids) for different events looks good. However, this bot is not a dedicated executable - it's part of the bigger MM2, and I would like to see a more universal architecture. Just as an idea I imagine the following notifications configuration:
> 
> ```json
> {
>     "telegram": {
>        "api_key": "key",
>        "api_url": "URL",
>        "default_chat_id": "default_chat_id",
>        "custom_chat_ids": {
>           "maker_bot": "id_for_maker_bot",
>           "swap_events": "id_for_swap_events",
>           "error_messages": "id_for_error_messages",
>        },
>     }
> }
> ```
> 
> How much effort do you think it will take to refactor it this way?
> 
> Also, `let msg = "simple_market_maker_bot successfully started";` is not extensible. I guess we won't send such notifications in the plain text e.g. to WS or gRPC, which we will hopefully have in the near future.
> 
> ```
> enum MakerBotEvent {
> ...
> }
> ```
> 
> will look much better.

Implemented, cfg is a bit different:

```
"message_service_cfg": {
    "telegram": {
      "api_key": "api_key",
      "chat_registry": {
        "default": "586216033",
        "maker_bot": "586216033",
        "swap_events": "586216033"
      }
    }
  }
```